torch/csrc/jit/script/logging.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <memory>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+namespace torch {
+namespace jit {
+namespace logging {
+
+class LoggerBase {
+ public:
+  TORCH_API virtual void addStatValue(
+      const std::string& stat_name,
+      float val) = 0;
+  TORCH_API virtual std::unordered_map<std::string, float> getCounters()
```

Comment:
Maybe getCounterValue is a better interface, to avoid aggregating/copying the entire logger when we want to get a single counter